### PR TITLE
[usage] only reset to max spendingLimit

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Reset usage to the spending limit at most.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/gitpod-db/go/cost_center.go [hunk 1]
```diff
@@ -169,17 +169,17 @@ func (c *CostCenterManager) UpdateCostCenter(ctx context.Context, newCC CostCent
 		return CostCenter{}, status.Errorf(codes.InvalidArgument, "Unknown attribution entity %s", string(attributionID))
 	}
 
-	// Downgrading from stripe
-	if existingCC.BillingStrategy == CostCenter_Stripe && newCC.BillingStrategy == CostCenter_Other {
+	// Transitioning into free plan
+	if existingCC.BillingStrategy != CostCenter_Other && newCC.BillingStrategy == CostCenter_Other {
```
Comment (easyCZ) [inline components/gitpod-db/go/cost_center.go:173]: Should this updated logic be covered by tests?

**Review by easyCZ — APPROVED** [review]
LGTM % is there a missing test for the updated logic?

/hold

File: components/gitpod-db/go/cost_center.go (hunk 1 above)
Comment (svenefftinge) [reply components/gitpod-db/go/cost_center.go:173]: The logic didn't change as there are only two different strategies. The point here is that it is not about not stripe but about going free (from anywhere). I'm working on a follow-up PR, where I introduce a third strategy for chargebee plans in cancellation mode. That PR comes with a test for this.

**Pull request merged**